Assets/Scripts/Abilities/AbstractAbility.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public abstract class AbstractAbility
+{
+    protected string _animatorID;
+    protected Rigidbody2D _body;
+    protected Animator _animator;
+    protected bool _active;
+    protected float _energyCost;
+
+    public static event Action<float> AbilityTriggered;
+
+    public AbstractAbility(string animatorID, Animator animator, Rigidbody2D body, float energyCost)
+    {
+        _animatorID = animatorID;
+        _animator = animator;
+        _body = body;
+        _active = false;
+        _energyCost = energyCost;
+    }
+
+    public abstract void Initialize(GameObject obj);
+    public abstract void Trigger(InputAction.CallbackContext ctxt);
```

Comment:
This is fine for now but I think there's a chance it'd be better to not directly rely on the input system here since a lot of abilities will end up colliding with each for input keys. There might be a way to make input systems dynamic though, so we change to a "jetpack" input mode when the character state changes.